io_scene_godot/converters/mesh.py

```diff
@@ -115,6 +116,79 @@ def export_object_link_material(escn_file, export_settings, mesh_object,
                 )
 
 
+def get_applicable_modifiers(obj, export_settings):
+    ignore_modifiers = []
+    if not export_settings['use_mesh_modifiers']:
+        return []
+    if "ARMATURE" in export_settings['object_types']:
+        ignore_modifiers.append(bpy.types.ArmatureModifier)
+    ignore_modifiers = tuple(ignore_modifiers)
+    return [m for m in obj.modifiers if not isinstance(m, ignore_modifiers)
+            and m.show_viewport]
+
+
+def record_modifier_config(obj):
+    modifier_config_cache = []
+    for mod in obj.modifiers:
+        modifier_config_cache.append(mod.show_viewport)
+    return modifier_config_cache
+
+def restore_modifier_config(obj, modifier_config_cache):
+    for i, mod in enumerate(obj.modifiers):
+        mod.show_viewport = modifier_config_cache[i]
+
+def has_shape_keys(object_data):
+    return (hasattr(object_data, "shape_keys") and
+            object_data.shape_keys is not None)
+
+def get_bl_rna_properties(obj):
+    """Property serialization helper"""
+    values = []
+    for k in obj.bl_rna.properties.keys()[1:]:
+        values.append(get_bl_prop_value(obj.id_data, getattr(obj, k)))
+    return tuple(values)
+
+NON_IMMUTABLE = (mathutils.Color, mathutils.Euler, mathutils.Matrix,
+                 mathutils.Quaternion, mathutils.Vector)
+
+def get_bl_prop_value(id_data, prop):
+    """Property serialization helper"""
+    if isinstance(prop, NON_IMMUTABLE):
+        return prop[:]
+    if isinstance(prop, bpy.types.bpy_prop_array):
+        return tuple(prop)
+    if isinstance(prop, bpy.types.bpy_prop_collection):
+        return tuple([get_bl_prop_value(id_data, item) for item in prop])
+    if isinstance(prop, bpy.types.bpy_struct):
+        # if bpy_struct relates to parent object (ie, a propertygroup or
+        # similar structs containing properties) then it's embedded data
+        # we want to traverse, not a pointer to another object.
+        # NOTE: If I'm wrong about any of this, expect bugs!
+        if prop.id_data == id_data:
+            return get_bl_rna_properties(prop)
+    return prop
+
+class MeshResourceKey:
+    """Produces a key based on an object's data, its enabled modifiers
+    and their properties."""
+    # pylint: disable=protected-access
+    def __init__(self, obj, export_settings):
```

Comment:
Here we may have to include the Godot resource name in the Key construction, because the same mesh can be used as ArrayMesh and CollisionShape at the same time. For more information, see the attached patch and the test case physics.escn
[0002-Include-Godot-Resource-Name-in-to-MeshResourceKey.patch.zip](https://github.com/godotengine/godot-blender-exporter/files/3270110/0002-Include-Godot-Resource-Name-in-to-MeshResourceKey.patch.zip)